src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -295,128 +296,55 @@ MmapPerfEvent ConsumeMmapPerfEvent(PerfEventRingBuffer* ring_buffer,
 
 StackSamplePerfEvent ConsumeStackSamplePerfEvent(PerfEventRingBuffer* ring_buffer,
                                                  const perf_event_header& header) {
-  // We expect the following layout of the perf event:
-  //  struct {
-  //    struct perf_event_header header;
-  //    u64 sample_id;          /* if PERF_SAMPLE_IDENTIFIER */
-  //    u32 pid, tid;           /* if PERF_SAMPLE_TID */
-  //    u64 time;               /* if PERF_SAMPLE_TIME */
-  //    u64 stream_id;          /* if PERF_SAMPLE_STREAM_ID */
-  //    u32 cpu, res;           /* if PERF_SAMPLE_CPU */
-  //    u64 abi;                /* if PERF_SAMPLE_REGS_USER */
-  //    u64 regs[weight(mask)]; /* if PERF_SAMPLE_REGS_USER */
-  //    u64 size;               /* if PERF_SAMPLE_STACK_USER */
-  //    char data[size];        /* if PERF_SAMPLE_STACK_USER */
-  //    u64 dyn_size;           /* if PERF_SAMPLE_STACK_USER && size != 0 */
-  //  };
-  // Unfortunately, the value of `size` is not constant, so we need to compute the offsets by hand,
-  // rather than relying on a struct.
-
-  size_t offset_of_size =
-      offsetof(perf_event_stack_sample_fixed, regs) + sizeof(perf_event_sample_regs_user_all);
-  size_t offset_of_data = offset_of_size + sizeof(uint64_t);
-
-  uint64_t size = 0;
-  ring_buffer->ReadValueAtOffset(&size, offset_of_size);
-
-  size_t offset_of_dyn_size = offset_of_data + (size * sizeof(char));
-
-  uint64_t dyn_size = 0;
-  ring_buffer->ReadValueAtOffset(&dyn_size, offset_of_dyn_size);
+  perf_event_attr flags{};
 
-  perf_event_sample_id_tid_time_streamid_cpu sample_id;
-  ring_buffer->ReadValueAtOffset(&sample_id, offsetof(perf_event_stack_sample_fixed, sample_id));
+  flags.sample_type =
```

Comment:
Can I ask for a comment here about why these 3 are used for sample_type?

Reply:
Those are the fields used by `StackSamplePerfEvent`. Basically, the flags follow the return type. 

`StackSamplePerfEvent` needs time, pid, tid, those are returned by `SAMPLE_TYPE_TID_TIME_STREAMID_CPU`. It also needs regs, that's returned by `PERF_SAMPLE_REGS_USER`. And finally, it needs dyn_size, and data which are returned by `PERF_SAMPLE_STACK_USER`.

To me, they look clear. Should I add a comment about this in the code?

Reply:
Looking at `PerfRecordSample` makes it clear what each flag is responsible for.

Reply:
We agreed offline about a comment pointing to PerfEventOpen file and also adding a TODO about not having duplicate code.

Reply:
Added.